specification/sections/05-events.md

```diff
@@ -147,22 +155,25 @@ See [provider initialization](./02-providers.md#24-initialization), [setting a p
 
 ### Event handlers and context reconciliation
 
-Providers built to conform to the static context paradigm feature an additional `PROVIDER_CONTEXT_CHANGED` event, which is used to signal that the global context has been changed, and flags should be re-evaluated.
+Providers built to conform to the static context paradigm feature two additional events: `PROVIDER_CONTEXT_CHANGE_PENDING` and `PROVIDER_CONTEXT_CHANGED`.
+While the provider is reconciling its internal state (the `on context changed` function is running and not yet terminated), the SDK emits `PROVIDER_CONTEXT_CHANGE_PENDING` and transitions the provider into state `CONTEXT_PENDING`.
+This can be particularly useful for displaying loading indicators while the [evaluation context](./03-evaluation-context.md) is being reconciled.
+
+If the `on context changed` function terminates normally, the SDK emits (`PROVIDER_CONTEXT_CHANGED`) and transitions the provider into the `READY` state, otherwise it emits `PROVIDER_ERROR` and transitions the provider into `ERROR` state.
+The `PROVIDER_CONTEXT_CHANGED` is used to signal that the associated context has been changed, and flags should be re-evaluated.
```

Comment:
> If we change to `RECONCILED` we should also change to something like `PROVIDER_RECONCILED`.

We could, and I consider that as well, but this would be a breaking change to any of the implementing client providers (including web). I felt that this wasn't worth it, but I could be convinced otherwise.

Reply:
Ah no sorry I did not consider that this already existed.
I would also leave it then.